frontend/public/components/cluster-settings/related-objects.tsx

```diff
@@ -1,116 +1,171 @@
   const { t } = useTranslation();
-  return <EmptyBox label={t('public~Related objects')} />;
+  const columns = React.useMemo(() => {
+    return [
+      {
+        title: t('public~Name'),
+        id: columnIds[0].id,
+        sort: 'name',
+        props: {
+          ...cellIsStickyProps,
+          modifier: 'nowrap',
+        },
+      },
+      {
+        title: t('public~Resource'),
+        id: columnIds[1].id,
+        sort: 'resource',
+        props: {
+          modifier: 'nowrap',
+        },
+      },
+      {
+        title: t('public~Group'),
+        id: columnIds[2].id,
+        sort: 'group',
+        props: {
+          modifier: 'nowrap',
+        },
+      },
+      {
+        title: t('public~Namespace'),
+        id: columnIds[3].id,
+        sort: 'namespace',
+        props: {
+          modifier: 'nowrap',
+        },
+      },
+    ];
+  }, [t]);
+  return columns;
 };
 
-const RelatedObjects: React.FC<RelatedObjectsProps> = (props) => {
+const RelatedObjects: React.FC<RelatedObjectsProps> = ({ data }) => {
   const { findModel } = useModelFinder();
   const { t } = useTranslation();
-  const Header = () => [
-    {
-      title: t('public~Name'),
-      sortField: 'name',
-      transforms: [sortable],
-      props: { className: tableColumnClasses[0] },
-    },
-    {
-      title: t('public~Resource'),
-      sortField: 'resource',
-      transforms: [sortable],
-      props: { className: tableColumnClasses[1] },
-    },
-    {
-      title: t('public~Group'),
-      sortField: 'group',
-      transforms: [sortable],
-      props: { className: tableColumnClasses[2] },
-    },
-    {
-      title: t('public~Namespace'),
-      sortField: 'namespace',
-      transforms: [sortable],
-      props: { className: tableColumnClasses[3] },
+  const columns = useRelatedObjectsColumns();
+
+  const getObjectMetadata = React.useCallback(
+    (object: ClusterOperatorObjectReference): ResourceMetadata => {
+      return { name: object.name };
     },
-  ];
-  const customData = React.useMemo(
-    () => ({
-      findModel,
-    }),
-    [findModel],
+    [],
   );
+
+  const customRowData: RelatedObjectsRowData = {
+    findModel,
+  };
+
   return (
-    <PaneBody>
-      <Table
-        {...props}
-        Header={Header}
-        Row={Row}
-        customData={customData}
-        aria-label={t('public~Related objects')}
-        NoDataEmptyMsg={EmptyMessage}
+    <React.Suspense fallback={<div className="loading-skeleton--table" />}>
+      <ConsoleDataView<ClusterOperatorObjectReference, RelatedObjectsRowData, RelatedObjectsFilters>
+        label={t('public~Related objects')}
+        data={data}
+        loaded={true}
+        columns={columns}
+        initialFilters={initialFiltersDefault}
+        getObjectMetadata={getObjectMetadata}
+        getDataViewRows={(rowData, tableColumns) =>
+          getRelatedObjectsDataViewRows(rowData, tableColumns)
```

Comment:
Nit - could be done directly as `getDataViewRows={getClusterOperatorDataViewRows}`